core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreMessageHandler.java

```diff
@@ -84,17 +118,85 @@ public void handleUpdate(SchemaRegistryKey key,
                            TopicPartition tp,
                            long offset,
                            long timestamp) {
+    if (key.getKeyType() != SchemaRegistryKeyType.SCHEMA && value == null) {
+      // ignore non-schema tombstone
+      return;
+    }
+
     if (key.getKeyType() == SchemaRegistryKeyType.SCHEMA) {
       handleSchemaUpdate((SchemaKey) key,
           (SchemaValue) value,
           (SchemaValue) oldValue);
-    } else if (value == null) {
-      // ignore non-schema tombstone
     } else if (key.getKeyType() == SchemaRegistryKeyType.DELETE_SUBJECT) {
       handleDeleteSubject((DeleteSubjectValue) value);
     } else if (key.getKeyType() == SchemaRegistryKeyType.CLEAR_SUBJECT) {
       handleClearSubject((ClearSubjectValue) value);
     }
+    recordBackup(key, value, tp, timestamp);
+  }
+
+  private void recordBackup(SchemaRegistryKey key,
+                            SchemaRegistryValue value,
+                            TopicPartition tp,
+                            long timestamp) {
+    boolean ignoreSubject = false;
+    if (key instanceof SubjectKey) {
+      String subject = ((SubjectKey) key).getSubject();
+      if (subject != null) {
+        int index = subject.indexOf("_");
+        if (index != -1) {
+          String tenantRemovedSubject = subject.substring(index + 1);
+          String[] subjectsBlacklist = schemaRegistry.config()
+                  .getString(SchemaRegistryConfig.BACKUPS_SUBJECT_BLACKLIST)
+                  .split(" ");
+          ignoreSubject = Arrays.asList(subjectsBlacklist).contains(tenantRemovedSubject);
```

Comment:
We do not need to call asList during recordBackup, right? We can generate it before that as is it is static based on the property.

Reply:
yup, this is true